rust/src/lib/nispor/ip.rs

```diff
@@ -150,15 +150,44 @@ pub(crate) fn nmstate_ipv4_to_np(
             });
         }
     }
+
+    if let Some(nms_cur_iface) = nms_merged_iface.current.as_ref() {
+        if let (Some(nms_cur_ipv4), Some(nms_des_ipv4)) = (
+            &nms_cur_iface.base_iface().ipv4,
+            &nms_merged_iface.merged.base_iface().ipv4,
+        ) {
+            let des_ips: Vec<_> = nms_des_ipv4
+                .addresses
+                .as_deref()
+                .unwrap_or_default()
+                .iter()
+                .map(|addr| addr.ip)
+                .collect();
+
+            for nms_addr in
+                nms_cur_ipv4.addresses.as_deref().unwrap_or_default()
+            {
+                if !des_ips.contains(&nms_addr.ip) {
+                    np_ip_conf.addresses.push({
+                        let mut ip_conf = nispor::IpAddrConf::default();
+                        ip_conf.address = nms_addr.ip.to_string();
+                        ip_conf.prefix_len = nms_addr.prefix_length;
+                        ip_conf.remove = true;
+                        ip_conf
+                    });
+                }
+            }
```

Comment:
IMO you should compare entire InterfaceIpAddr here. Like:
```
let des_ips: Vec<_> = nms_des_ipv4.addresses.as_deref().unwrap_or_default();
for ... {
    if !des_ips.contains(&nms_addr) { ... }
}
```

Otherwise, if the current state contains `92.168.1.10/32 and the desired state contains 192.168.1.10/24 the old one won't be removed. Same for IPv6.

Reply:
Actually, after doing the change, that case may deserve a test to avoid a regression in the future.